### PR TITLE
[1LP][RFR]5.11 ec2 image upload

### PR DESCRIPTION
**Pull request by mmojzis**
Needs this wrapanapi PR https://github.com/RedHatQE/wrapanapi/pull/440 to be working.

### CODE REVIEW

File: cfme/utils/template/ec2.py [hunk 1]
```diff
@@ -49,25 +49,47 @@ def upload_image(self):
...
-                     fail_condition=False,
-                     delay=5,
-                     timeout='90m',
-                     message='Importing image to EC2')
-
-            ami_id = self.mgmt.get_image_id_if_import_completed(import_task_id)
-            ami = EC2Image(uuid=ami_id, system=self.mgmt)
-            ami.set_tag("Name", self.template_name)
-            return True
+        if self.stream == 'downstream-510z':
```
Comment (mmojzis) [inline cfme/utils/template/ec2.py]: Maybe create another method for 5.11+ and this condition would be in run method?

**Comment by dajoRH** [comment]
Lint report for commit 3ed239a948018cb143d3e27eea89030421e1c142:

`cfme/utils/template/ec2.py`:
- :heavy_exclamation_mark: Line 115:1: **W293** *blank line contains whitespace*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 69fc009be6e73c69946088c987c43b3ac15aa896:

`cfme/utils/template/ec2.py`:
- :heavy_exclamation_mark: Line 114:38: **W291** *trailing whitespace*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/utils/template/ec2.py [hunk 2]
```diff
@@ -49,25 +49,47 @@ def upload_image(self):
...
+                         message='Importing image to EC2')
+
+                snapshot_id = self.mgmt.get_snapshot_id_if_import_completed(import_task_id)
+                ami_result = self.mgmt.create_image_from_snapshot(self.template_name, snapshot_id)
+                ami_id = ami_result.get("ImageId")
+                ami = EC2Image(uuid=ami_id, system=self.mgmt)
+                ami.set_tag("Name", self.template_name)
+                return True
+
+            except Exception:
```
Comment (jaryn) [inline cfme/utils/template/ec2.py]: **required:** @mmojzis Please remove the `except Exception` from here and above or please do explain why you won't let the exception propagate. Letting it to propagate will lead to easier debugging. It is rarely a good idea to muffle exceptions. And even if we want to muffle some, we should specify which ones and not just simply catch all `Exception` instances.

https://realpython.com/the-most-diabolical-python-antipattern/

File: cfme/utils/template/ec2.py [hunk 3]
```diff
@@ -49,25 +49,47 @@ def upload_image(self):
...
+                                                        s3key=self.template_name,
+                                                        description=self.template_name)
+                wait_for(self.mgmt.get_snapshot_id_if_import_completed,
+                         func_args=[import_task_id],
+                         fail_condition=False,
+                         delay=5,
+                         timeout='90m',
+                         message='Importing image to EC2')
+
+                snapshot_id = self.mgmt.get_snapshot_id_if_import_completed(import_task_id)
```
Comment (jaryn) [inline cfme/utils/template/ec2.py]: Up until this the two branches are same.

File: cfme/utils/template/ec2.py [hunk 4]
```diff
@@ -49,25 +49,47 @@ def upload_image(self):
...
+                ami_result = self.mgmt.create_image_from_snapshot(self.template_name, snapshot_id)
+                ami_id = ami_result.get("ImageId")
+                ami = EC2Image(uuid=ami_id, system=self.mgmt)
```
Comment (jaryn) [inline cfme/utils/template/ec2.py]: From here down they seem to also be the same. I think you can de-dup this a bit.

File: cfme/utils/template/ec2.py (hunk 1 above)
Comment (mshriver) [reply cfme/utils/template/ec2.py]: I think its fine where it is, the logic is specific to the import step.

File: cfme/utils/template/ec2.py (hunk 2 above)
Comment (mshriver) [reply cfme/utils/template/ec2.py]: Good point @jaryn, this is propagating a poor pattern that was introduced before this PR.

Given the use and scope of these template upload scripts, we do want to diaper exceptions here - but not without logging them.

Addition of `logger.exception` call before returning would be sufficient to log what occurred while maintaining some of the logging done by the wrapper/decorator for these methods.

**Pull request merged**